include/clap/preset-discovery.h

```diff
@@ -0,0 +1,244 @@
+
+   /**
+    * Adds a feature to the preset.
+    * See plugin-features.h and preset-features.h
+    *
+    * The feature string is arbitrary, it is the indexer's job to remap it to its internal
+    * categorization and tagging system.
+    */
+   void(CLAP_ABI *add_feature)(const struct clap_preset_metadata_receiver *receiver,
+                               const char                                 *feature);
+
+} *clap_preset_metadata_receiver_t;
+
+enum clap_preset_discovery_flags {
+   // This location/collection is meant for storing factory presets, most likely read-only
+   CLAP_PRESET_DISCOVERY_IS_FACTORY_CONTENT = 1 << 0,
+
+   // This location/collection is meant for storing user created presets
+   CLAP_PRESET_DISCOVERY_IS_USER_CONTENT = 1 << 1,
+
+   // This location/collection is meant for demo presets, those are preset which may trigger
+   // some limitation in the plugin because they require additionnal features which the user
+   // needs to purchase or the content itself needs to be bought and is only available in
+   // demo mode.
+   CLAP_PRESET_DISCOVERY_IS_DEMO_CONTENT = 1 << 2,
+};
+
+// Defines a place in which to search for presets
+typedef struct clap_preset_location {
+   uint64_t flags; // see enum clap_preset_discovery_flags
+
+   // name of this location
+   char name[CLAP_NAME_SIZE];
+
+   // path to a directory or a file on the file system in which preset can be found
+   char path[CLAP_URI_SIZE];
+} clap_preset_location_t;
+
+// A collection, represent a collection of presets; it is will most often used to identify presets
+// which belongs to the same "sound pack".
+typedef struct clap_preset_collection_info {
+   uint64_t flags; // see enum clap_preset_discovery_flags
+
+   char name[CLAP_NAME_SIZE];               // name of this collection
+   char description[CLAP_DESCRIPTION_SIZE]; // reasonably short description of the collection
+   char homepage_url[CLAP_URI_SIZE];        // url to the pack's homepage
+   char image_uri[CLAP_URI_SIZE];           // may be an image on disk or from an http server
+} clap_preset_collection_info_t;
+
+// Describes a preset provider
+typedef struct clap_preset_provider_descriptor {
+   clap_version_t     clap_version; // initialized to CLAP_VERSION
+   const char        *id;
+   const char        *name;
+   const char        *vendor;
+   const char *const *plugin_ids; // null terminated array of plugin ids
+} clap_preset_provider_descriptor_t;
+
+// This interface isn't thread-safe.
+typedef struct clap_preset_provider {
+   const clap_preset_provider_descriptor_t *desc;
+
+   // Destroys the preset provider
+   void(CLAP_ABI *destroy)(const struct clap_preset_provider *provider);
+
+   // Retrives the path to a watch file.
+   // Whenever the given file is "touched", then the indexer shall invalidate all the data.
+   bool(CLAP_ABI *invalidation_watch_file)(const struct clap_preset_provider *provider,
```

Comment:
This isn't possible because the preset provider isn't a running service.
If you move content you could maybe touch a file on disk which the indexer could monitor?